density/density.py

```diff
@@ -493,5 +493,27 @@ def map():
     # Render template has an SVG image whose colors are changed by % full
     return render_template('map.html', locations=locations)
 
+@app.route('/upload', methods = ['POST'])
+def upload():
+    """ Accept POST requests from CUIT to add new data to the server """
+
+    # This is stored in local settings and is the way we verify uploads.
+    secret_key = request.args['key']
+    if secret_key != app.config['UPLOAD_KEY']:
+        return 'Please include a valid API key.', 401
+
+    json_data = request.get_json()
+    if 'data' not in json_data:
+        return 'Please include data to upload.', 400
+
+    density_data = json_data['data']
+
+    db_failed = db.insert_density_data(g.cursor, density_data)
```

Comment:
Maybe we should check for other conditions? Like make sure that the data contains all rooms we support?